### ISSUE TITLE
seg fault possibly caused by or related to collocation grid cache build

### ISSUE THREAD

**Issue by hokru** [psi4/psi4#1366]
With a recent master (1.3a2.dev179, Git: Rev {master} c46660a) the following calculation fails at least 1/10 times. AMD Ryzen, Intel2018 compiler, compiled again conda libraries.

`for i in  {1..10}; do psi4dev -s /scratch/ -n 6 water.in out ; done`

water.in:
```memory 4 GiB
molecule water {
O            0.000000000000    -0.000000000000     0.065656839916  
H           -0.000000000000     0.758561000000    -0.521010847646  
H            0.000000000000    -0.758561000000    -0.521010847646  
}
set {
scf_type df
basis def2-QZVP
dft_spherical_points 590
dft_radial_points 100
}
energy('wb97X-V')
```
The segfault's backtrace contains `psi.VBase.build_collocation_cache`.
Turning the collocation grid _off_ fixes the issue.

**Comment by Diazonium** [comment]
What model of Ryzen CPU are you using? Some specimens of the 1st generation Ryzen family (1200-1800X) have exhibited random segfaults while running certain programs. (most notably, compiling GCC using GCC).

**Comment by hokru** [comment]
Ryzen 5 1600, but probably not from the faulty early batches.

Anyway I see the same behaviour on an Intel E5-2667v4.

**Comment by PeterKraus** [comment]
I can confirm it on other relatively new Intel processors. Didn't try on my ancient AMD's, but I think we can safely assume it's Psi4 and not some hardware quirk.

**Comment by hokru** [comment]
using intel-gdb I am getting a segfault at:
`0x00002aaac4d3f06e in std::local_Rb_tree_decrement (__x=0x2aaaae4b3ff0 <psi::SphericalGrid::lebedev_mapping_+8>)`

Possibly multi-threading related, since I have not seen an error for single-threaded _yet_.

**Comment by dgasmith** [comment]
That’s a curious stack trace. Can I get the full one?

**Comment by hokru** [comment]
for gdb? There isn't much more
```
Starting program: /home/kruse/miniconda3/bin/python /usr/qc/psi4fork.bin/bin/psi4 -n4 water.in
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[New Thread 0x2aaaf01ac780 (LWP 10083)]
[New Thread 0x2aab0f5ae800 (LWP 10084)]
[New Thread 0x2aab37400880 (LWP 10085)]

Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00002aaac4d3f06e in std::local_Rb_tree_decrement (__x=0x2aaaae4b3ff0 <psi::SphericalGrid::lebedev_mapping_+8>)
    at /opt/conda/conda-bld/compilers_linux-64_1534514838838/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/libstdc++-v3/src/c++98/tree.cc:97
97      /opt/conda/conda-bld/compilers_linux-64_1534514838838/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/libstdc++-v3/src/c++98/tree.cc: No such file or directory.
```

I needed to reproduce the backtrace error from normal execution. Not always a useful crash, too. But there seem to be different errors.
Intel: [stacktrace.txt](https://github.com/psi4/psi4/files/2610768/stacktrace.txt)
AMD:  [stacktrace_amd.txt](https://github.com/psi4/psi4/files/2610774/stacktrace_amd.txt)

**Comment by dgasmith** [comment]
The stack traces shown indicate the issue is `clear_collection_cache` not `build`. Seems like this error is all over the place. Do you have SCF iterations before the crash?

I have a guess as to the error, I will make a PR.

**Comment by hokru** [comment]
> Do you have SCF iterations before the crash?

No. It crashes setting everything up.

And yes, the error is not reliably found in one place.

**Comment by JonathonMisiewicz** [comment]
@hokru Are you still encountering this? I don't think we checked if #1387 fixed this.

**Comment by hokru** [comment]
I believe it was fixed by that PR.

**Comment by JonathonMisiewicz** [comment]
Close the issue, then?

**Comment by hokru** [comment]
fixed by #1387
